### PR TITLE
aws - kinesis Analytics V2, subnet filter and delete action

### LINKED ISSUE #6688: Add support for Kinesis Analytics V2
**Is your feature request related to a problem? Please describe.**
Make sure all kinesis analytics services are in a private subnet

**Describe the solution you'd like**
Add a new resource type for KinesisAnalyticsV2 including tagging, delete and subnet filter. Link to boto3 docs: https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/kinesisanalyticsv2.html#KinesisAnalyticsV2.Client.describe_application

**Describe alternatives you've considered**
NA

**Additional context**
NA

### PR DESCRIPTION
**Pull request by PratMis**
Closes #6688

### CODE REVIEW

File: c7n/resources/kinesis.py [hunk 1]
```diff
@@ -278,6 +279,48 @@ def augment(self, resources):
...
+        service = "kinesisanalyticsv2"
+        enum_spec = ('list_applications', 'ApplicationSummaries', None)
+        detail_spec = ('describe_application', 'ApplicationName',
+                       'ApplicationName', 'ApplicationDetail')
+        name = "ApplicationName"
+        arn = id = "ApplicationARN"
+        arn_type = 'application'
+        universal_taggable = object()
+        cfn_type = 'AWS::KinesisAnalyticsV2::Application'
+
```
Comment (kapilt) [inline c7n/resources/kinesis.py:295]: add `permission_prefix = "kinesisanalytics"`

Comment (PratMis) [reply c7n/resources/kinesis.py:296]: That fixed it. Thanks @kapilt

**Comment by PratMis** [comment]
Looks like CI is not able to import docker

**Review by kapilt — APPROVED** [review]
lgtm, thanks

**Pull request merged**